### PR TITLE
[to DEV] Refactor ConstructorState and MethodState

### PR DESCRIPTION
**Pull request by LucianoPC**
Refactor class ConstructorState to inherit class MethodState

Issue:
MethodState and ConstructorState are very similar #69

### CODE REVIEW

**Comment by rodrigosiqueira** [comment]
We will adopt LGPL instead of GPL, if you agree with this I will proceed with your MR, otherwise I will refuse it. Do you agree with this?

**Comment by LucianoPC** [comment]
Sure

File: lib/kuniri/state_machine/OO_structured_fsm/constructor_state.rb [hunk 1]
```diff
@@ -1,66 +1,25 @@
...
 module StateMachine
 
   module OOStructuredFSM
 
     # Class responsible for handling constructor state.
-    class ConstructorState < OOStructuredState
+    class ConstructorState < MethodState
 
-      @language
+      CLASS_TYPE = "constructor"
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/constructor_state.rb:11]: I like this idea! However, I believe that isn't a good idea to set a string directly to the attribute as you did. I recommend you to create some constants in token_state_machine.rb and just use it here. 
Something like this:

`CLASS_TYPE ||= StateMachice::CLASS_TYPE`

File: lib/kuniri/state_machine/OO_structured_fsm/method_state.rb [hunk 2]
```diff
@@ -7,10 +7,21 @@ module OOStructuredFSM
...
+      METHOD_TYPE = "function"
+
       @language
 
-      def initialize(pLanguage)
+      @state
+      @classType
+      @methodType
+
+      def initialize(pLanguage, params = {})
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/method_state.rb:19]: change: params to pParams
Add:
- Add a small documentation about each parameter (use yardoc).
- Add a unit test for check those parameters.

File: lib/kuniri/state_machine/OO_structured_fsm/method_state.rb [hunk 3]
```diff
@@ -31,25 +42,25 @@ def class_capture
...
       def repetition_capture
-        @language.flagFunctionBehaviour = StateMachine::METHOD_STATE
+        @language.flagFunctionBehaviour = @state
         @language.set_state(@language.repetitionState)
       end
 
       # @see OOStructuredState
       def execute(pElementFile, pLine)
-        methodElement = @language.methodHandler.get_function(pLine)
+        methodElement = @language.send("#{@classType}Handler").send("get_#{@methodType}", pLine)
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/method_state.rb:57]: Cool! :)
Just, remember to break line. We just use 80 char columns.

File: lib/kuniri/state_machine/OO_structured_fsm/method_state.rb [hunk 4]
```diff
@@ -31,25 +42,25 @@ def class_capture
...
 
         if (methodElement)
           lastIndex = pElementFile.classes.length - 1 # We want the index
           methodElement.comments = @language.string_comment_to_transfer
           @language.string_comment_to_transfer = ""
-          pElementFile.classes[lastIndex].add_method(methodElement)
+          pElementFile.classes[lastIndex].send("add_#{@classType}", methodElement)
```
Comment (rodrigosiqueira) [inline lib/kuniri/state_machine/OO_structured_fsm/method_state.rb:63]: To long line, please, break it.

**Comment by rodrigosiqueira** [comment]
Thanks for your contribution, your code is really cool. Just apply some few changes, then I will merge it into dev.

**Pull request merged**